api-list.go

```diff
@@ -77,7 +77,7 @@ func (c Client) ListBuckets() ([]BucketInfo, error) {
 //
 func (c Client) ListObjects(bucketName, objectPrefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectInfo {
 	// Allocate new list objects channel.
-	objectStatCh := make(chan ObjectInfo, 1000)
+	objectStatCh := make(chan ObjectInfo, 1)
```

Comment:
If this is a buffer channel of 1 entry then the application for loop will read one extra entry unless it does an explicit break (but if we have to mandate a break then this chan can be buffered for 1000 entries too). It should be unbuffered channel
objectStatCh := make(chan ObjectInfo)